core/arch/x86/decode_fast.c

```diff
@@ -1282,7 +1316,9 @@ decode_cti(dcontext_t *dcontext, byte *pc, instr_t *instr)
      * on every return. */
     uint rip_rel_pos;
 #endif
-    int sz = decode_sizeof(dcontext, pc, &prefixes _IF_X64(&rip_rel_pos));
+    byte vex = 0, vex_mm = 0;
+    int sz = decode_sizeof_internal(dcontext, pc, &prefixes, &vex,
+                                    &vex_mm _IF_X64(&rip_rel_pos));
```

Comment:
I don't know, it seems simpler, cleaner, and my intuition says it's more efficient to leave decode_sizeof as it is and put in the change I originally suggested, just checking for the vex prefixes in the loop on line 1342 which is already checking prefixes.  What was your logic for doing it this way instead?  Adding extra conditionals and stores to decode_sizeof seems to add more code complexity and probably overhead (have to measure to say for sure) than adding some case statements to the existing loop (or change to if-else if compiler isn't doing that for opt).

Reply:
Well the rationale was since decode_sizeof already looks at the prefixes and decodes vex_mm, why duplicate it and do it again. So I moved it into an internal function. I don't have a strong preference, the original suggestion will work as well, except it doesn't check the ordering of the prefixes (but neither does decode_sizeof).